HLTrigger/special/src/HLTHcalLaserMisfireFilter.cc

```diff
@@ -0,0 +1,131 @@
+  minFracDiffHBHELaser_ = config.getParameter<double>("minFracDiffHBHELaser");
+  minFracHFLaser_       = config.getParameter<double>("minFracHFLaser");
+  minADCHBHE_           = config.getParameter<int>("minADCHBHE");
+  minADCHF_             = config.getParameter<int>("minADCHF"),
+  testMode_             = config.getUntrackedParameter<bool>("testMode",false);
+
+  inputTokenHBHE_ = consumes<HBHEDigiCollection>(inputHBHE_);
+  inputTokenHF_   = consumes<QIE10DigiCollection>(inputHF_);
+
+ }
+
+HLTHcalLaserMisfireFilter::~HLTHcalLaserMisfireFilter() { }
+ 
+void HLTHcalLaserMisfireFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("InputHBHE",edm::InputTag("source"));
+  desc.add<edm::InputTag>("InputHF",edm::InputTag("source"));
+  desc.add<int>("minADCHBHE",10);
+  desc.add<int>("minADCHF",10);
+  desc.add<double>("minFracDiffHBHELaser",0.3);
+  desc.add<double>("minFracHFLaser",0.3);
+  desc.addUntracked<bool>("testMode",false);
+  descriptions.add("hltHcalLaserMisfireFilter",desc);
+}
+
+bool HLTHcalLaserMisfireFilter::filter(edm::StreamID, edm::Event& iEvent,
+				       const edm::EventSetup& iSetup) const {
+
+  edm::Handle<HBHEDigiCollection>     hbhe_digi; 
+  iEvent.getByToken(inputTokenHBHE_,  hbhe_digi);
+
+  edm::Handle<QIE10DigiCollection>    hf_digi; 
+  iEvent.getByToken(inputTokenHF_,    hf_digi);
+
+  // Count digis in good, bad RBXes.  ('bad' RBXes see no laser signal)
+  double badrbxfracHBHE(0), goodrbxfracHBHE(0), goodrbxfracHF(0);
+  int NbadHBHE = 72*3;             // 3 bad RBXes, 72 channels each
+  int NgoodHBHE= 2592*2-NbadHBHE;  // remaining HBHE channels are 'good'
+  int NgoodHF  = 864*4;
+
+  for (auto hbhe = hbhe_digi->begin();  hbhe != hbhe_digi->end(); ++ hbhe){
+    const HBHEDataFrame digi = (const HBHEDataFrame)(*hbhe);
+    HcalDetId myid=(HcalDetId)digi.id();
+    bool isbad(false); // assume channel is not bad
+
+    bool passCut(false);
+    int  maxdigiHB(0);
+    for (int i=0; i<digi.size(); i++) 
+      if(digi.sample(i).adc() > maxdigiHB) maxdigiHB = digi.sample(i).adc();
+    if (maxdigiHB > minADCHBHE_) passCut = true;
+
+    if ( myid.subdet()==HcalBarrel && myid.ieta()<0 ) {
+      if (myid.iphi()>=15 && myid.iphi()<=18) isbad=true;
+      else if (myid.iphi()>=27 && myid.iphi()<=34) isbad=true;
+    }
+
+    if (passCut) {
+      if (isbad) { 
+	badrbxfracHBHE += 1.;
+      } else goodrbxfracHBHE += 1.;
+    }
+  }
+  goodrbxfracHBHE /= NgoodHBHE;
+  badrbxfracHBHE  /= NbadHBHE;
+  
+  for (auto hf = hf_digi->begin(); hf != hf_digi->end(); ++hf) {
+    const QIE10DataFrame digi = (const QIE10DataFrame)(*hf);
+    bool passCut(false);
+    int  maxdigiHF(0);
+    for (int i=0; i<digi.samples(); i++) 
+      if(digi[i].adc() > maxdigiHF) maxdigiHF = digi[i].adc();
+    if (maxdigiHF > minADCHF_) passCut = true;
+
+    if (passCut) {
+      goodrbxfracHF += 1.;
+    }
+  }
+  goodrbxfracHF /= NgoodHF;
```

Comment:
for now, it seems this should be called NHF. as there is no concept of bad HF in the code?